card: read headphone availability from input device

peat-psuwit opened the pull request:
Recent Android devices start to provide headphone availability via input
device instead of h2w switch. This renders droid-extcon useless.

This commit introduces droid-extevdev, a simple code that will read
headphone availability from the input device, using libevdev as an
abstraction layer. This means the code now depends on libevdev (but it
can be made optional later on if needed).

This should make headphone availability works on newer Android devices
without having to resort to h2w kernel driver.

peat-psuwit committed:
[debian] add libevdev-dev as a build dependency

Flohack74 commented:
@fredldotme didn´t you work on a similar thing for suzu?

fredldotme commented:
> @fredldotme didn´t you work on a similar thing for suzu?

My solution would've been in telepathy-ofono, before we found out that headset detection was to be found in this big patch file where h2w and virtual voice streams for pulseaudio come from.

peat-psuwit commented:
@Flohack74 What he does on suzu is that he adds h2w driver to kernel, which makes droid-extcon works. However, I believe that patching all kernels with the h2w driver isn't a scalable solution, so I decided to develop this patch.

@fredldotme This is not from Canonical's patch. This has been written from scratch by me, referencing how it works on Android.

fredldotme commented:
@peat-psuwit what Flohack and I talked about was a solution similar to this PR.

peat-psuwit commented:
Pending Jenkinsfile update in #4.

Flohack74 commented:
Yeah I agree if we can have a solution without kernel patching this would be great.

mariogrip commented:
@peat-psuwit Merged, could you push/rebase with the updated jenkins file? :)

peat-psuwit committed:
card: read headphone availability from input device

Recent Android devices start to provide headphone availability via input
device instead of h2w switch. This renders droid-extcon useless.

This commit introduce droid-extevdev, a simple code that will read
headphone availability from the input device, using libevdev as an
abstraction layer. This means the code now depends on libevdev (but it
can be made optional later on if needed).

This should make headphone availability works on newer Android devices
without having to resort to h2w kernel driver.

peat-psuwit commented:
Now ready to be review!

fredldotme reviewed (approved):
LGTM

UniversalSuperBox commented:
Ready for merge, but we still need to discuss whether we want to put it after OTA-12.

mariogrip commented:
> Ready for merge, but we still need to discuss whether we want to put it after OTA-12.

This is for pulseaudio-modules-droid-24 (halium-7.1) so none of the *core* devices use this anyway, so should be fine to merge.

mariogrip reviewed (approved):
LGTM :+1:
